test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -23,15 +25,41 @@
 #include "framework/integration_framework/iroha_instance.hpp"
 #include "framework/integration_framework/test_irohad.hpp"
 #include "framework/result_fixture.hpp"
+#include "interfaces/iroha_internal/transaction_batch_factory_impl.hpp"
+#include "interfaces/iroha_internal/transaction_batch_parser_impl.hpp"
 #include "interfaces/permissions.hpp"
 #include "module/shared_model/builders/protobuf/block.hpp"
 #include "module/shared_model/builders/protobuf/proposal.hpp"
+#include "module/shared_model/validators/validators.hpp"
+#include "multi_sig_transactions/transport/mst_transport_grpc.hpp"
+#include "network/impl/async_grpc_client.hpp"
 #include "synchronizer/synchronizer_common.hpp"
 
 using namespace shared_model::crypto;
 using namespace std::literals::string_literals;
 
-static size_t kToriiPort = 11501;
+using AlwaysValidProtoCommonObjectsFactory =
+    shared_model::proto::ProtoCommonObjectsFactory<
+        shared_model::validation::AlwaysValidFieldValidator>;
+using ProtoTransactionFactory = shared_model::proto::ProtoTransportFactory<
+    shared_model::interface::Transaction,
+    shared_model::proto::Transaction>;
+using AbstractTransactionValidator =
+    shared_model::validation::AbstractValidator<
+        shared_model::interface::Transaction>;
+using AlwaysValidTransactionValidator =
+    shared_model::validation::AlwaysValidModelValidator<
+        shared_model::interface::Transaction>;
+
+static std::string kLocalHost = "127.0.0.1";
+static constexpr size_t kDefaultToriiPort = 11501;
+static constexpr size_t kDefaultInternalPort = 50541;
+
+template <size_t default_port>
+static size_t getNextPort() {
+  static size_t increment = 0;
+  return default_port + (++increment);
```

Comment:
Not a real situation, but I think it might be nice to additionally check for overflow (since ports are usually lesser than ~65k) and throw exception if any